### DISCUSSION TITLE
Is there a sample for separated CLI and VSCode projects?

### DISCUSSION THREAD

**Discussion by ymc9** [eclipse-langium/langium#980] [Q&A] [answered]
Hi,

I'm thinking of having a CLI (parser/linker/validator) decoupled with VSCode extension. They seem to be coupled to some extent, but I haven't tried to refactor yet. Just wondering if there's already a reference implementation somewhere. Thanks!

**Comment by msujew** [comment, marked as answer]
Hey @ymc9,

there is! The Langium monorepo itself is built that way. The `langium` package contains the language, while the `langium-vscode` extension only bundles it into an extension.

**Reply by ymc9** [reply]
Awesome! I'll look into it. Thank you!
